ElementX/Sources/Services/Timeline/RoomTimelineController.swift

```diff
@@ -133,65 +139,69 @@ class RoomTimelineController: RoomTimelineControllerProtocol {
             return
         }
         
-        mediaProvider.loadImageFromSource(source) { [weak self] result in
-            guard let self = self else {
+        switch await mediaProvider.loadImageFromSource(source) {
+        case .success(let image):
+            guard let index = self.timelineItems.firstIndex(where: { $0.id == timelineItem.id }),
+                  var item = self.timelineItems[index] as? ImageRoomTimelineItem else {
                 return
             }
             
-            if case let .success(image) = result {
-                guard let index = self.timelineItems.firstIndex(where: { $0.id == timelineItem.id }),
-                      var item = self.timelineItems[index] as? ImageRoomTimelineItem else {
-                    return
-                }
-                
-                item.image = image
-                self.timelineItems[index] = item
-                self.callbacks.send(.updatedTimelineItem(timelineItem.id))
-            }
+            item.image = image
+            self.timelineItems[index] = item
+            self.callbacks.send(.updatedTimelineItem(timelineItem.id))
```

Comment:
I think this is good now (maybe caught by Sonarcloud anyway)
```suggestion
            timelineItems[index] = item
            callbacks.send(.updatedTimelineItem(timelineItem.id))
```

Reply:
Same for the rest below too :)